demo/shared/src/main/scala/games/input/Keyboard.scala

```diff
@@ -0,0 +1,141 @@
+package games.input
+
+import java.io.Closeable
+
+object Keyboard {
+  private[games] type KeyMapper[T] = BiMap[Key, T]
+}
+
+sealed abstract class Key
+
+object Key {
+  case object Space extends Key
```

Comment:
For performance reasons, you should really make these `final val Space = 1` etc.
Dispatching on that many `case objects` will be very slow compared to swithing on an integer. Both on JVM and JS, but even more so on JS.